src/VisualStudio/CSharp/Impl/EventHookup/EventHookupSessionManager_EventHookupSession.cs

```diff
@@ -111,17 +112,18 @@ public void Cancel()
                         _cancellationTokenSource.Token,
                         TaskScheduler.Default);
 
-                    var continuedTask = this.GetEventNameTask.SafeContinueWith(t =>
+                    var continuedTask = this.GetEventNameTask.SafeContinueWithFromAsync(
+                        async t =>
                         {
-                            AssertIsForeground();
+                            await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, _cancellationTokenSource.Token);
                             if (t.Result != null)
                             {
                                 commandHandler.EventHookupSessionManager.EventHookupFoundInSession(this);
                             }
                         },
                         _cancellationTokenSource.Token,
-                        TaskContinuationOptions.OnlyOnRanToCompletion,
-                        ForegroundThreadAffinitizedObject.CurrentForegroundThreadData.TaskScheduler);
+                        TaskContinuationOptions.OnlyOnRanToCompletion | TaskContinuationOptions.ExecuteSynchronously,
+                        TaskScheduler.Default);
```

Comment:
another example of this repeated pattern.

Reply:
➡️ Covered under the discussion in https://github.com/dotnet/roslyn/pull/28781#discussion_r208378244